appengine/pond/js/avatar.js

```diff
@@ -34,22 +34,46 @@ goog.require('BlocklyInterface');
  * Class for a avatar.
  * @param {string} name Avatar's name.
  * @param {string|!Function} code Avatar's code, or generator.
- * @param {!Blockly.utils.Coordinate} startLoc Start location.
  * @param {?number} startDamage Initial damage to avatar (0-100, default 0).
- * @param {!Pond.Battle} battle The battle featuring the avatar.
+ * @param {?Blockly.utils.Coordinate} opt_startLoc Start location.
  * @constructor
  */
-Pond.Avatar = function(name, code, startLoc, startDamage, battle) {
+Pond.Avatar = function(name, code, startDamage, opt_startLoc) {
   this.name = name;
   this.code_ = code;
-  this.startLoc_ = startLoc;
+  this.startLoc_ = opt_startLoc;
   this.startDamage_ = startDamage || 0;
-  this.battle_ = battle;
+  this.battle_ = null;
   this.loc = new Blockly.utils.Coordinate();
-  this.reset();
   console.log(this + ' loaded.');
 };
 
+/**
+ * Given a name create the default player.
+ * @param {string} name The name of the default player.
+ */
+Pond.Avatar.createDefaultAvatar = function(name) {
```

Comment:
The avatar names should be the translated message retrieved with "BlocklyGames.getMsg", So you  need to call that here.